### PR TITLE
Update to AGP 3.0.0a8

### PR DESCRIPTION
**Pull request by jrodbx**
- Bump to Gradle 4.0-M1
- Bump to AGP 3.0.0-alpha8
- Update dependency configuration syntax
- Tell Robolectric to include Android resources in tests
- Update plugin usage of variant api

### CODE REVIEW

**Comment by jrodbx** [comment]
```
* What went wrong:
A problem occurred configuring project ':sample:library'.
> Failed to notify project evaluation listener.
   > com.android.build.gradle.api.BaseVariant.getOutputs()Ljava/util/List;
```

This error happens locally as well.  It seems that :sample:library has a buildscript dependency on the previously published version of butterknife-gradle-plugin, which fails with 3.0.0-alpha8 due to a variant api change, so we have a 🐔 & 🥚.  Essentially, this build won't pass unless a new version of the plugin is published.

To test locally, I did the following:
1) In gradle/gradle-mvn-push.gradle, update urls to local repo
```
  repository(url: mavenLocal().url) {
  snapshotRepository(url: mavenLocal().url) {
```
2) Build and install updated plugin to local repository
```
  ./gradlew :butterknife-gradle-plugin:uploadArchives
```
3) In sample/library/build.gradle, add mavenLocal() to buildscript.repositories
```
  buildscript {
    repositories {
      ...
      mavenLocal()
    }
  }
```
4) Also, in sample/library/build.gradle, update plugin version to local SNAPSHOT
```
  dependencies {
    classpath "com.jakewharton:butterknife-gradle-plugin:SNAPSHOT"
  }
```
5) Finally, build and test
```
  ./gradlew :sample:app:build
```

**Comment by JakeWharton** [comment]
Just disable the sample from settings.gradle

**Comment by jrodbx** [comment]
works for me.  updated.

File: butterknife-compiler/build.gradle [hunk 1]
```diff
@@ -13,14 +13,14 @@ for (File file : sdkHandler.sdkLoader.repositories) {
 }
 
 dependencies {
-  compile project(':butterknife-annotations')
-  compile deps.auto.common
-  compile deps.javapoet
+  implementation project(':butterknife-annotations')
+  implementation deps.auto.common
+  api deps.javapoet
```
Comment (JakeWharton) [inline butterknife-compiler/build.gradle]: This is implementation too, no? We don't expose these types in the public API.

Comment (jrodbx) [inline butterknife-compiler/build.gradle]: maybe?  i changed it to `api` in response to this error:
```
> Task :butterknife:compileDebugUnitTestJavaWithJavac
error: Bad service configuration file, or exception thrown while constructing Processor object: javax.annotation.processing.Processor: Provider butterknife.compiler.ButterKnifeProcessor could not be instantiated: java.lang.NoClassDefFoundError: com/squareup/javapoet/ClassName
```

Comment (JakeWharton) [inline butterknife-compiler/build.gradle]: This is because of how we're running the compiler in the tests. Just going to leave it for simplicity.

Comment (jrodbx) [inline butterknife-compiler/build.gradle]: 👍

**Comment by denizmveli** [comment]
@jrodbx this error is still occurring for me using butterknife 8.8.0 in one of our library projects. It's using `alpha9`.

```
* What went wrong:
A problem occurred configuring project ':mobile'.
> Could not resolve all dependencies for configuration ':mobile:kapt'.
   > A problem occurred configuring project ':player'.
      > Failed to notify project evaluation listener.
         > com.android.build.gradle.api.BaseVariant.getOutputs()Ljava/util/List
```

I've cleaned all caches and re-run. Is there something obvious I'm missing?

**Comment by JakeWharton** [comment]
Use 9.0.0-SNAPSHOT

**Comment by Frank1234** [comment]
Could you update the snapshot to beta1? "Gradle sync failed: Could not find com.android.tools.build:gradle:3.0.0-alpha9."

**Comment by JakeWharton** [comment]
That's an issue on your side unless Google removed alpha9 from their repo
(which seems unlikely).

**Comment by jrodbx** [comment]
Just tested and 3.0.0-alpha9 is still available. @Frank1234 did you include google() in your buildscript.repositories?  That's a necessary step now if you happen to be migrating from 2.3.3 to 3.0.0*.

**Comment by Frank1234** [comment]
thanks @jrodbx, that's it

File: settings.gradle [hunk 2]
```diff
@@ -5,7 +5,7 @@ include ':butterknife-gradle-plugin'
 include ':butterknife-lint'
 include ':butterknife-integration-test'
 
-include ':sample:app'
-include ':sample:library'
+//include ':sample:app'
```
Comment (jaredsburrows) [inline settings.gradle]: @jrodbx @JakeWharton Did you mean to leave this commented out?

Comment (jrodbx) [inline settings.gradle]: yep.  see https://github.com/JakeWharton/butterknife/pull/1026#issuecomment-318657996

**Comment by jaredsburrows** [comment]
Thanks. I was looking into solutions earlier to use the plugin module
directly (buildSrc module, jar directly, snapshot, etc). The buildSrc
solution is the only one that solves it entirely, I might make a PR later.

**Comment by AndSync** [comment]
remove  apply plugin: 'com.jakewharton.butterknife' may be worked

File: butterknife-compiler/build.gradle (hunk 1 above)
Comment (TWiStErRob) [reply butterknife-compiler/build.gradle]: @JakeWharton `implementation deps.javapoet; testRuntimeOnly deps.javapoet` so the consumers are not affected, and it reflects the intent?

Comment (JakeWharton) [reply butterknife-compiler/build.gradle]: Sure. Send a PR

**Pull request merged**